File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r = 0,len(height)-1
        res=0
        while r>l:
            a=(r-l) * min(height[l],height[r]) #area of a rec. l*w
            res=max(res, a)
            if height[r]>height[l]:
                l+=1
            else:
                r-=1 # we can increase l or right for when l and r are equal right or r<l
        return res        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxA=INT_MIN;

        int l=0;
        int r=height.size()-1;

        while(l<r){
            int area = (r-l) * min(height[r], height[l]);
            maxA = max(area, maxA); 
            if(height[r]>=height[l]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxA;
    }
};